### DISCUSSION TITLE
Creating Additional Dashboards

### DISCUSSION THREAD

**Discussion by mengelsen** [jertel/vuegraf#162] [Q&A] [answered]
If I create additional Dashboards or Members (within the InfluxDB UI) can those changes be preserved if the Vuegraf and InfluxDB containers gets removed and recreated?

**Comment by jertel** [comment, marked as answer]
As long as you followed the Vuegraf installation guide, and specifically, bound a persistent directory into the InfluxDB container for the permanent storage of InfluxDB data, then yes, you can destroy the containers and recreate them without losing data.

**Reply by mengelsen** [reply]
That makes sense. I followed the [Installation Guide](https://github.com/jertel/vuegraf?tab=readme-ov-file#setup).

```
mkdir -p /home/pi/influxdb2
docker run -v /home/pi/influxdb2:/var/lib/influxdb2 -p 8086:8086 -e INFLUXD_SESSION_LENGTH=432000 --name influxdb influxdb
```
...
```
docker run --name vuegraf -d -v /home/pi/vuegraf:/opt/vuegraf/conf jertel/vuegraf
```
The contents of the `/home/pi/influxdb2` (local directory) do match the contents of the directory within the (InfluxDB) Docker container.
<img width="527" alt="Screenshot 2024-08-15 at 11 33 36 AM" src="https://github.com/user-attachments/assets/34c11f38-c27a-45bb-b580-3ee2bc58ba6e">

However, within the `influxdb2` directory, I only see a copy of **your** `influx_dashboard.json` file. 
If I created a new dashboard within the InfluxDB UI, would "my dashboard" be saved (as a .json file) somewhere within the `/home/pi/influxdb2/engine/data/` directory?
Or are dashboards contained within the `influxd.sqlite` file?

**Reply by jertel** [reply]
They are stored in sqlite. Remember during the installation guide that you imported that json file into InfluxDB.

**Reply by mengelsen** [reply]
That all makes perfect sense. Just to be on the safe side, I also found a way to export my dashboard to a .json file.
<img width="1096" alt="Screenshot 2024-08-15 at 1 25 38 PM" src="https://github.com/user-attachments/assets/b13a6187-fce6-4994-9270-dd57858f5eba">

**Side Note**: would you consider updating your Installation Instructions? There are (2) references to `docker run`.
Can you please append the following launch argument to the `docker run` command:
`-restart always`
I think others will find it helpful if their InfluxDB and Vuegraf containers auto-start on boot.

**Reply by mengelsen** [reply]
Updated `docker run` commands:
```
sudo docker run -d -v /home/<<username_HERE>>/influxdb2:/var/lib/influxdb2 -p 8086:8086 -e INFLUXD_SESSION_LENGTH=432000 --restart always --name influxdb influxdb
sudo docker run --name vuegraf -d -v /home/<<username_HERE>>/vuegraf:/opt/vuegraf/conf jertel/vuegraf --restart always
```

**Reply by mengelsen** [reply]
The --restart argument works w/o issue on influxdb. But `vuegraf` doesn't currently accept --restart

```
pi@raspberrypi:~ $ sudo docker logs vuegraf
usage: vuegraf.py [-h] [-v] [-d] [--historydays HISTORYDAYS] [--resetdatabase]
                  configFilename
vuegraf.py: error: unrecognized arguments: --restart
```

**Reply by jertel** [reply]
I could, but this is getting more into _productionalizing_ a server, and configuring the container to restart any time it crashes or the server reboots is just a small piece of what's required to do this correctly.

Perhaps a note at the end of the getting started guide that reminds them there are additional steps necessary for making this configuration fault tolerant, such as configuring the container to always restart, backing up their InfluxDB on a frequent basis, log rollover management, OS alerts to an admin when InfluxDB crashes, checking for low disk space on the host, setting up calendar reminders for host OS updates and associated kernel reboots, updating Vuegraf and Influx on a schedule, etc.

Also, move the `--restart always` after the `docker run` words to avoid the Vuegraf error you pasted.
